manim/animation/indication.py

```diff
@@ -340,9 +617,116 @@ def interpolate_submobject(
         )
 
 
+class WiggleOutThenIn(Wiggle):
+    """Deprecated. Use :class:`~.Wiggle` instead."""
+
+    def __init__(*args, **kwargs):
+        logger.warning(
+            "WiggleOutThenIn has been deprecated in favor of Wiggle. Please use Wiggle instead!"
+        )
+        super().__init(*args, **kwargs)
+
+
 class TurnInsideOut(Transform):
+    """Deprecated. Use :code:`mobject.animate.become(mobject.copy().reverse_points())` instead if you have to."""
+
     def __init__(self, mobject: "Mobject", path_arc: float = TAU / 4, **kwargs) -> None:
+        logger.warning(
+            "TurnInsideOut has been deprecated. Please stop using TurnInsideOut!"
+        )
         super().__init__(mobject, path_arc=path_arc, **kwargs)
 
     def create_target(self) -> "Mobject":
         return self.mobject.copy().reverse_points()
+
+
+class Circumscribe(Succession):
+    """Draw a temporary line surrounding the mobject.
+
+    Parameters
+    ----------
+    mobject
+        The mobject to be circumscribed.
+    shape
+        The shape with which to surrond the given mobject. Should be either "rectangle" or "circle"
+    fade_in
+        Whether to make the surrounding shape to fade in. It will be drawn otherwise.
+    fade_out
+        Whether to make the surrounding shape to fade out. It will be undrawn otherwise.
+    time_width
+        The time_width of the drawing and undrawing. Gets ignored if either `fade_in` or `fade_out` is `True`.
+    buff
+        The distance between the surrounding shape and the given mobject.
+    color
+        The color of the surrounding shape.
+    run_time
+        The duration of the entire animation.
+    kwargs : Any
+        Additional arguments to be passed to the :class:`~.Succession` constructor
+
+    Examples
+    --------
+
+    .. manim:: UsingCircumscribe
+
+        class UsingCircumscribe(Scene):
+            def construct(self):
+                lbl = Tex(r"Circum-\\\\scribe")
+                self.add(lbl)
+                self.play(Circumscribe(lbl))
+                self.play(Circumscribe(lbl, "circle"))
+                self.play(Circumscribe(lbl, fade_out=True))
+                self.play(Circumscribe(lbl, time_width=2))
+                self.play(Circumscribe(lbl, "circle", True))
+
+    """
+
+    def __init__(
+        self,
+        mobject: Mobject,
+        shape: Union[str, Type] = "rectangle",
```

Comment:
I must say, I really hate using harcoded string for as arguments for that. Why not accpeting only vmobjects ?

Reply:
Allowing only the classes Rectangle and Circle as parameters would give the impression other VMobject subclasses might work as well. If if you want to use shape=Circle that works as well...

Reply:
At the end of the day it's the same as accepting only two strings "rectangle" and "circle", so why not ? I don't think it will be misleading, given that it is written in the docs